### ISSUE TITLE
100+ running git commands alert firing regularly

### ISSUE THREAD

**Issue by keegancsmith** [sourcegraph/sourcegraph#13683]
`[Cloud] [CRITICAL] gitserver: 100+ running git commands (signals load)` is firing very regularly. This issue is about investigating and fixing.

- [x] Silence / adjust alert (maybe just add for duration to avoid noise)
- [ ] Find root cause
- [ ] Fix

cc @sourcegraph/cloud @pecigonzalo

**Comment by keegancsmith** [comment]
I investigated this briefly by looking in honeycomb at a specific alert. All the git commands where git logs against a specific repo. It seems most of the `git logs` where repeated. So something is acting up. My best guess this is search related. Either diff/commit search or a graphql resolver poorly caching information.

Example from alert that fired at Sep 7th 2020 3:28pm UTC: (5:28pm SAST)

Grouping by repository we see it dominated by mattermost (69k requests, vs 2k for the next top repo): https://ui.honeycomb.io/sourcegraph/datasets/gitserver-exec/result/3ct5aeRRwZo

![image](https://user-images.githubusercontent.com/187831/92445985-7571fa80-f1b5-11ea-8d07-becccf214be6.png)


Filtering to just the mattermost repo and grouping by cmd,args gives us lots of git logs, many repeated: https://ui.honeycomb.io/sourcegraph/datasets/gitserver-exec/result/mXc76gnrXpR

![image](https://user-images.githubusercontent.com/187831/92445763-1dd38f00-f1b5-11ea-80bd-506397e700bc.png)

Next Steps: It is unclear why we are doing this. I would try correlate this with search requests. I looked in our logs but we don't seem to be logging anything about the request. We should introduce a honeycomb dataset for search. Alternatively we should have this in our logs.
